appcell-resources/js/socialgraph.js

```diff
@@ -222,82 +222,91 @@ sg.dispRelationLinkExtCell = function(json, relName, relBoxName) {
+              profObj.DisplayName = profData.DisplayName;
+              profObj.Description = profData.Description;
               if (profData.Image) {
-                  imageSrc = profData.Image;
+                  profObj.Image = profData.Image;
               }
           }
-          sg.appendRelationLinkExtCell(extUrl, dispName, description, imageSrc, extRelID, true);
+          sg.appendRelationLinkExtCell(extUrl, profObj, extRelObj, true);
       }).fail(function() {
-          var dispName = sg.getName(extUrl);
-          var description = "";
-          var imageSrc = cm.notImage;
-          sg.appendRelationLinkExtCell(extUrl, dispName, description, imageSrc, extRelID, true);
+          sg.appendRelationLinkExtCell(extUrl, profObj, extRelObj, true);
       });
 };
-sg.appendRelationLinkExtCell = function(url, dispName, description, imageSrc, extRelID, delFlag) {
+sg.appendRelationLinkExtCell = function(url, profObj, extRelObj, delFlag) {
     var aTag = $('<a>', {
         class: 'allToggle',
         onClick: 'sg.createExtCellProfile(this)'
     });
     var tempHTML = [
         '<table class="table-fixed">',
             '<tr>',
-                '<td rowspan="2" style="width: 25%;"><img class="image-circle" src="' + imageSrc + '" alt="user"></td>',
-                '<td>' + dispName + '</td>',
+                '<td rowspan="2" style="width: 25%;"><img class="image-circle" src="' + profObj.Image + '" alt="user"></td>',
+                '<td>' + profObj.DisplayName + '</td>',
             '</tr>',
             '<tr>',
-                '<td><p class="ellipsisText"><font color="LightGray">' + description + '</font></p></td>',
+                '<td><p class="ellipsisText"><font color="LightGray">' + profObj.Description + '</font></p></td>',
             '</tr>',
         '</table>'
     ].join("");
     aTag
         .data('url', url)
-        .data('dispName', dispName)
-        .data('description', description)
-        .data('imageSrc', imageSrc)
+        .data('dispName', profObj.DisplayName)
+        .data('description', profObj.Description)
+        .data('imageSrc', profObj.Image)
         .append(tempHTML);
     var firstCell = $('<td>', {
         style: 'width: 90%;'
     }).append($(aTag));
 
     var html='';
     if (delFlag) {
-        var splitID = extRelID.split("-");
-        var relName = splitID[1];
         var boxName = "[main]";
-        if (splitID.length > 2) {
-            boxName = splitID[2];
+        if (extRelObj.BoxName) {
+            boxName = extRelObj.BoxName;
         }
-        html = '<a class="del-button list-group-item" style="top:25%" href="#" onClick="sg.dispDelExtCellRelationModal(\'' + url + '\',\'' + relName + '\',\'' + boxName + '\');return(false)" data-i18n="Del"></a>';
+        var anotherTag = $('<a>', {
+            class: 'del-button list-group-item', 
+            style: 'top:25%;', 
+            onClick: 'sg.dispDelExtCellRelationModal(this)',
+            'data-i18n': 'Del'
```

Comment:
I think data-i18n is not allowed, the initialization only allow HTML standard attributes.

Reply:
For non-standard attributes, you can specify it with a single quote.